openedx/core/djangoapps/user_api/accounts/serializers.py

```diff
@@ -247,6 +250,40 @@ def validate_social_links(self, value):
             raise serializers.ValidationError("The social_links field must consist of unique social platforms.")
         return value
 
+    def validate_date_of_birth(self, value):
+        """
+        Method which validates date of birth that should be in the past.
+        """
+
+        value = self.convert_empty_to_None(value)
+
+        if value is not None and value > date.today():
+            raise AccountValidationError({
+                "date_of_birth": {
+                    "developer_message": "Value is not valid for field date_of_birth",
+                    "date_of_birth": [u"The date_of_birth field should be in the past."],
+                    "user_message": "This value is invalid.",
+                }
+            })
+
+        return value
+
+    def validate_phone(self, value):
+        """
+        Method which validates the phone field that should contain only numbers and brackets.
+        """
+
+        if value is not None and "_" in value:
```

Comment:
> only numbers and brackets

I prefer regexp to validate the restriction. What is your opinion around this?

Reply:
We have validation on the frontend side. I think it is not necessary now.

Reply:
On the frontend side? Really? Never trust frontend's validation 🙂
If we do not validate, it's fine, but your docstring totally confusing me
> Method which validates the phone field that should contain only numbers and brackets.

Are we going to write truth here?)

Reply:
Add phone validation by regexp.